src/node/http.ts

```diff
@@ -1,938 +1,253 @@
-   * allow the user to authenticate only once. This will use the highest level
-   * domain (e.g. `coder.com` over `test.coder.com` if both are specified).
-   */
-  private getCookieDomain(host: string): string | undefined {
-    const idx = host.lastIndexOf(":")
-    host = idx !== -1 ? host.substring(0, idx) : host
-    if (
-      // Might be blank/missing, so there's nothing more to do.
-      !host ||
-      // IP addresses can't have subdomains so there's no value in setting the
-      // domain for them. Assume anything with a : is ipv6 (valid domain name
-      // characters are alphanumeric or dashes).
-      host.includes(":") ||
-      // Assume anything entirely numbers and dots is ipv4 (currently tlds
-      // cannot be entirely numbers).
-      !/[^0-9.]/.test(host) ||
-      // localhost subdomains don't seem to work at all (browser bug?).
-      host.endsWith(".localhost") ||
-      // It might be localhost (or an IP, see above) if it's a proxy and it
-      // isn't setting the host header to match the access domain.
-      host === "localhost"
-    ) {
-      logger.debug("no valid cookie doman", field("host", host))
-      return undefined
-    }
-
-    this.proxyDomains.forEach((domain) => {
-      if (host.endsWith(domain) && domain.length < host.length) {
-        host = domain
+    // Send the request off to be handled by Express.
+    ;(app as any).handle(req, res, () => {
+      if (!req._ws_handled) {
+        socket.destroy(new Error("Not found"))
       }
     })
+  })
+}
 
-    logger.debug("got cookie doman", field("host", host))
-    return host ? `Domain=${host}` : undefined
-  }
-
-  /**
-   * Return a response if the request should be proxied. Anything that ends in a
-   * proxy domain and has a *single* subdomain should be proxied. Anything else
-   * should return `undefined` and will be handled as normal.
-   *
-   * For example if `coder.com` is specified `8080.coder.com` will be proxied
-   * but `8080.test.coder.com` and `test.8080.coder.com` will not.
-   *
-   * Throw an error if proxying but the user isn't authenticated.
-   */
-  public maybeProxy(route: ProviderRoute, request: http.IncomingMessage): HttpResponse | undefined {
-    // Split into parts.
-    const host = request.headers.host || ""
-    const idx = host.indexOf(":")
-    const domain = idx !== -1 ? host.substring(0, idx) : host
-    const parts = domain.split(".")
-
-    // There must be an exact match.
-    const port = parts.shift()
-    const proxyDomain = parts.join(".")
-    if (!port || !this.proxyDomains.has(proxyDomain)) {
-      return undefined
-    }
-
-    // Must be authenticated to use the proxy.
-    route.provider.ensureAuthenticated(request)
-
-    return {
-      proxy: {
-        port,
-      },
-    }
+/**
+ * Patch Express routers to handle web sockets and async routes (since we have
+ * to patch `get` anyway).
+ *
+ * Not using express-ws since the ws-wrapped sockets don't work with the proxy
```

Comment:
Let's link to the upstream issue for this.

Reply:
I suspect they don't have any plans on letting raw sockets pass through since the library seems focused on specifically wrapping the sockets with `ws`. Still, maybe they could make it optional. The library was last updated in 2018 so idk how active they still are.

I removed the bit about the wildcard since I think that can actually be worked around although I think the method used here is still superior.